File: frontend/src/pages/dashboard/employees/types/employee.ts
export interface Employee {
  id: number;
  employee_code: string;
  full_name: string;
  email: string;
  phone: string | null;
  designation: string;
  company_id: number;
  team_id: number | null;
  department_id: number;
  manager_id: number | null;
  joining_date: string;
  salary: number;
  status: string;
  created_at: string;
}

export interface EmployeeCreate {
  employee_code: string;
  full_name: string;
  email: string;
  phone?: string;
  designation: string;
  company_id: number;
  team_id?: number | null;
  department_id: number;
  manager_id?: number | null;
  joining_date: string;
  salary: number;
  status?: string;
}

export interface EmployeeUpdate {
  full_name?: string;   
  phone?: string;
  team_id?: number | null;
  designation?: string;
  department_id?: number;
  manager_id?: number | null;
  joining_date?: string;
  salary?: number;
  status?: string;
}

File: backend/app/schemas/employee.py
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, EmailStr

class EmployeeCreate(BaseModel):
    employee_code: str
    full_name: str
    email: EmailStr
    phone: str | None = None
    designation: str
    company_id: int
    team_id: int | None = None
    department_id: int
    manager_id: int | None = None
    joining_date: date
    salary: Decimal
    status: str = "ACTIVE"

class EmployeeUpdate(BaseModel):
    full_name: str | None = None
    phone: str | None = None
    team_id: int | None = None
    designation: str | None = None
    department_id: int | None = None
    manager_id: int | None = None
    joining_date: date | None = None
    salary: Decimal | None = None
    status: str | None = None

class EmployeeResponse(BaseModel):
    id: int
    employee_code: str
    full_name: str
    email: EmailStr
    phone: str | None
    designation: str
    company_id: int
    team_id: int | None
    department_id: int
    manager_id: int | None
    joining_date: date
    salary: Decimal
    status: str
    created_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )


